README.md

```diff
@@ -192,205 +141,129 @@ data SwapDatum
 
-#### What if two arbitragers compete for the same swap?
-Recall the contrived example above. What would happen if Charlie and Mike try to arbitrage it at the same time?
+Full benchmarking details can be found in the [Benchmarks](Benchmarks.md) file.
 
-1. Charlie and Mike both build their transactions since the UTxOs still exist.
-2. Charlie and Mike submit their transaction at the same time.
-3. Charlie's is added to a block first.
-4. When Mike's transaction is then picked to go into a block, the UTxOs no longer exist. The transaction fails without needing to run the swap contracts.
 
-Since Mike's transaction will fail without needing to run the swap script, Mike's collateral is safe. Further, the more available swaps there are, the less likely these "collisions" will occur.
+## Future Directions
+Features and considerations for future versions of Cardano-Swaps:
 
-#### If Cardano-Swaps reaches mass adoption, won't TVL on Cardano go down?
+### Offer-Based Swap Addresses
+Currently, swap addresses are unique for every trading pair. This necessitates many distinct Beacon UTxOs (and associated deposits), even if users are offering the same asset in exchange for varying assets. Future versions of Cardano-Swaps will instead utilize "Offer-based" swap addresses, which are unique only to the asset being offered. The desired "exchange" asset will be specified in the datum of every "Swap" UTxO at the address. This will dramatically lower Beacon UTxO deposit fees, as only one such UTxO will be needed per "Offer" asset, per user. This will also improve querying efficiency; instead of having to lookup a specific trading pair, users can lookup all swap addresses offering Token "ABC", and then filter through the desired exchange asset(s).
 
-Yes. Yes it will. TVL is a silly metric. It is a measure of who can be most inefficient with DeFi capital.
+> **Note**
+> This feature will require the introduction of a new Beacon Token.
 
-#### I have an idea for how to improve the protocol. Where should I go?
 
-Feel free to use either the Issue section or the Discussion section on the Cardano-Swaps repo. Any feedback that can improve the protocol is welcome!
+### Redundant Script Executions
+Currently, plutus validator scripts must be executed once for *every* UTxO being consumed from a script address (this is the case for Plutus as a whole, not just Cardano-Swaps). If multiple "Swap" UTxOs from the same address are consumed as inputs, the validator must execute for each of them, even though their spending logic is exactly the same. These executions are "redundant", in a sense, resulting in a (low severity) composability bottleneck. Addressing this would likely require significant changes to Cardano's transaction structure, which is unnecessary at this point in time.  [Present and future discussions with the IOG team](https://github.com/cardano-foundation/CIPs/pull/418) will help determine if this feature is indeed warranted.
```

Comment:
I'm not a fan of this section because it still gives off a vibe that redundant executions are holding back this design when in fact they are not. IMO This design can handle all the demand that it might experience for the foreseeable future. Composability is already good enough even with the other protocols. 

Unless the demand actually takes off and the dex is clearly bottlenecked, I can't see how the cost to fix the redundant executions is worth it. I think this section should basically say:
"The current design is more than performant enough for the demand we expect. If at some point the demand exceeds expectations, fixing the redundant executions could give a slight boost to performance. But as of now, they appear harmless."